sogs/model/bot.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+from .. import crypto, db, config
+from ..db import query
+from ..web import app
+from .exc import BadPermission, PostRateLimited
+from .. import utils
+from ..omq import omq_global
+from .user import User
+from .room import Room
+from .message import Message
+from .filter import SimpleFilter
+from .exc import InvalidData
+
+from typing import Optional, List, Union
+import time
+
+
+class Bot:
```

Comment:
I'd probably drag this whole file out of here and put it in its own `utils/bots/filter_bot.py` file. If its detached completely from the core then users will be able to copy and paste this Bot and make adjustments to turn it into whatever they want.

The bot itself wont want to touch anything like the database or models because its purely working though an OMQ connection and whatever API we define so the bot can communicate with the mule.